addons/website_sale_delivery/models/sale_order.py

```diff
@@ -75,7 +76,11 @@ def _get_delivery_methods(self):
         # Following loop is done to avoid displaying delivery methods who are not available for this order
         # This can surely be done in a more efficient way, but at the moment, it mimics the way it's
         # done in delivery_set method of sale.py, from delivery module
-        carrier_ids = DeliveryCarrier.sudo().search(
+        user = request.env.user
+        if user.has_group('base.group_website_publisher') and user.has_group('sales_team.group_sale_manager'):
+            carrier_ids = DeliveryCarrier.search([]).ids
```

Comment:
@rim-odoo  sales_team.group_sale_manager has already rights of manager for delivery.carrier https://github.com/odoo/odoo/blob/master/addons/delivery/security/ir.model.access.csv#L4  so, not use sudo()